File: new/atoms/is-valid-pin.test.mjs
import { isValidPin } from './is-valid-pin.mjs';
const CASES = [[["\"123456782\""],"false"],[["\"039217369\""],"false"],[["{\"amount\":100}"],"false"],[["{\"payments\":[{\"amount\":100},{\"amount\":50}]}"],"false"],[["{\"name\":\"כהן\",\"phone\":\"0501234567\"}"],"false"],[["[{\"amount\":100}]"],"false"],[["[\"2026-08-24\"]"],"false"],[["3.14"],"false"],[["1000"],"true"],[["2026"],"true"],[["\"\""],"false"],[["\"אבג\""],"false"]];
const de = (s) => s === '"__undef__"' ? undefined : JSON.parse(s);
let f = 0;
for (const [args, want] of CASES) { const got = JSON.stringify(isValidPin(...args.map(de))); if (got !== want) { console.error('✗ ' + args + ' ⇒ ' + got + ' ≠ ' + want); f = 1; } }
if (f) process.exit(1); console.log('✓ is-valid-pin: ' + CASES.length + ' הקלטות-Golden — ירוק');

File: new/dart-maor/is-valid-pin_test.dart
// זהב · is-valid-pin — assert-ים = בדיוק דוגמאות-החוזה של is-valid-pin.test.mjs (JS≡Dart).
// הערכים הומרו לערכי-Dart ילידיים (המקבילים ל-JSON.parse של קלטי-ה-Golden).
import 'is-valid-pin.dart';

void main() {
  assert(isValidPin('123456782') == false); // 9 ספרות
  assert(isValidPin('039217369') == false); // 9 ספרות
  assert(isValidPin({'amount': 100}) == false); // מפה
  assert(isValidPin({'payments': [{'amount': 100}, {'amount': 50}]}) == false);
  assert(isValidPin({'name': 'כהן', 'phone': '0501234567'}) == false);
  assert(isValidPin([{'amount': 100}]) == false); // מערך-מפות
  assert(isValidPin(['2026-08-24']) == false); // מקפים
  assert(isValidPin(3.14) == false); // נקודה
  assert(isValidPin(1000) == true); // 4 ספרות
  assert(isValidPin(2026) == true); // 4 ספרות
  assert(isValidPin('') == false); // ריק
  assert(isValidPin('אבג') == false); // לא-ספרות
  print('✓ is-valid-pin: 12 הקלטות-Golden — Dart≡JS ירוק');
}
